packages/remix-dev/cli/commands.ts

```diff
@@ -303,7 +303,8 @@ export async function generateEntry(
   let defaultEntryClient = path.resolve(defaultsDirectory, "entry.client.tsx");
   let defaultEntryServer = path.resolve(
     defaultsDirectory,
-    ctx?.remixConfig.ssr === false
+    ctx?.remixConfig.ssr === false &&
+      ctx?.remixConfig.future.unstable_singleFetch !== true
```

Comment:
Reveal the streaming `entry.server` in SPA mode when single fetch is enabled.  

I wonder if there's any good way to detect if a user is currently using `renderToString` in their own `entry.server` and they flip on single fetch.  From what I recall the error isn't super obvious and React sticks it in a `<template>` at the end of the HTML but no console errors get logged.

Reply:
@jacob-ebey is going to look into a check at built time for this using the `entry.server` AST (separate PR)